internal/resource/obclusteroperation/obclusteroperation_task.go

```diff
@@ -0,0 +1,248 @@
+			obcluster.Spec.OBServerTemplate.Storage.RedoLogStorage.Size = *mutation.RedoLogStorage
+		}
+	case strings.ToLower(string(constants.ClusterOpTypeModifyStorageClass)):
+		if m.Resource.Spec.ModifyStorageClass == nil {
+			return errors.New("ModifyStorageClass is empty")
+		}
+		mutation := m.Resource.Spec.ModifyStorageClass
+		if mutation.DataStorage != "" {
+			obcluster.Spec.OBServerTemplate.Storage.DataStorage.StorageClass = mutation.DataStorage
+		}
+		if mutation.LogStorage != "" {
+			obcluster.Spec.OBServerTemplate.Storage.LogStorage.StorageClass = mutation.LogStorage
+		}
+		if mutation.RedoLogStorage != "" {
+			obcluster.Spec.OBServerTemplate.Storage.RedoLogStorage.StorageClass = mutation.RedoLogStorage
+		}
+	case strings.ToLower(string(constants.ClusterOpTypeSetParameters)):
+		if m.Resource.Spec.SetParameters == nil {
+			return errors.New("setParameters is empty")
+		}
+		newParamMap := make(map[string]string)
+		for _, v := range m.Resource.Spec.SetParameters {
+			newParamMap[v.Name] = v.Value
+		}
+		existingMap := make(map[string]struct{})
+		for i, v := range obcluster.Spec.Parameters {
+			if val, ok := newParamMap[v.Name]; ok {
+				obcluster.Spec.Parameters[i].Value = val
+			}
+			existingMap[v.Name] = struct{}{}
+		}
+		for k, v := range newParamMap {
+			if _, ok := existingMap[k]; !ok {
+				obcluster.Spec.Parameters = append(obcluster.Spec.Parameters, apitypes.Parameter{Name: k, Value: v})
+			}
+		}
+	}
+	if m.Resource.Spec.Force {
+		obcluster.Status.Status = clusterstatus.Running
+		obcluster.Status.OperationContext = nil
+	}
+	err = m.Client.Patch(m.Ctx, obcluster, client.MergeFrom(origin))
+	if err != nil {
+		m.Logger.Error(err, "Failed to patch obcluster")
+		return err
+	}
+
+	err = m.waitForOBClusterToBeStatus(obcfg.GetConfig().Time.DefaultStateWaitTimeout, func(status string) bool {
+		return status != clusterstatus.Running
+	})
+	if err != nil {
+		return errors.New("Timeout to wait for cluster to be operating")
+	}
+	return nil
+}
+
+func WaitForCluster(m *OBClusterOperationManager) tasktypes.TaskError {
+	timeout := obcfg.GetConfig().Time.DefaultStateWaitTimeout
+	if m.Resource.Spec.Type == constants.ClusterOpTypeModifyStorageClass {
+		timeout = obcfg.GetConfig().Time.ServerDeleteTimeoutSeconds
+	}
+	err := m.waitForOBClusterToBeStatus(timeout, func(status string) bool {
+		return status == clusterstatus.Running
+	})
+	if err != nil {
+		return errors.New("Timeout to wait for cluster to be running")
+	}
+	return nil
+}
+
+func RestartServers(m *OBClusterOperationManager) tasktypes.TaskError {
+	obcluster := &v1alpha1.OBCluster{}
+	err := m.Client.Get(m.Ctx, types.NamespacedName{
+		Namespace: m.Resource.Namespace,
+		Name:      m.Resource.Spec.OBCluster,
+	}, obcluster)
+	if err != nil {
+		m.Logger.Error(err, "Failed to find obcluster")
+		return err
+	}
+	if obcluster.Annotations[oceanbaseconst.AnnotationsSupportStaticIP] != "true" {
+		return errors.New("RestartOBServers only support static ip")
```

Comment:
the error message can be changed to 'RestartOBServers requires static ip support'